main.tf

```
@@ -61,3 +60,34 @@ resource "github_team_membership" "all" {
   username = each.key
   role     = "member"
 }
+
+resource "github_organization_settings" "settings" {
+  count = length(var.settings) > 0 ? 1 : 0
+
+  billing_email                                                = var.settings.billing_email
+  company                                                      = try(var.settings.company, null)
+  blog                                                         = try(var.settings.blog, null)
+  email                                                        = try(var.settings.email, null)
+  twitter_username                                             = try(var.settings.twitter_username, null)
+  location                                                     = try(var.settings.location, null)
+  name                                                         = try(var.settings.name, null)
+  description                                                  = try(var.settings.description, null)
+  has_organization_projects                                    = try(var.settings.has_organization_projects, true)
+  has_repository_projects                                      = try(var.settings.has_repository_projects, true)
+  default_repository_permission                                = try(var.settings.default_repository_permission, "read")
+  members_can_create_repositories                              = try(var.settings.members_can_create_repositories, true)
+  members_can_create_public_repositories                       = try(var.settings.members_can_create_public_repositories, true)
+  members_can_create_private_repositories                      = try(var.settings.members_can_create_private_repositories, true)
+  members_can_create_internal_repositories                     = try(var.settings.members_can_create_internal_repositories, null)
+  members_can_create_pages                                     = try(var.settings.members_can_create_pages, true)
+  members_can_create_public_pages                              = try(var.settings.members_can_create_public_pages, true)
+  members_can_create_private_pages                             = try(var.settings.members_can_create_private_pages, false)
```

should all be false as we can assume we use IaC for this

```suggestion
  members_can_create_repositories                              = try(var.settings.members_can_create_repositories, false)
  members_can_create_public_repositories                       = try(var.settings.members_can_create_public_repositories, false)
  members_can_create_private_repositories                      = try(var.settings.members_can_create_private_repositories, false)
  members_can_create_internal_repositories                     = try(var.settings.members_can_create_internal_repositories, false)
  members_can_create_pages                                     = try(var.settings.members_can_create_pages, false)
  members_can_create_public_pages                              = try(var.settings.members_can_create_public_pages, false)
  members_can_create_private_pages                             = try(var.settings.members_can_create_private_pages, false)
```

---

should `members_can_create_internal_repositories` not be set to null? it's available in GH Enterprise only according to the documentation?

---

changed in 84aa087